Pipeline/DeployUCDComponentVersion/ucd-deploy.groovy

```diff
@@ -0,0 +1,192 @@
+		new SSLConnectionSocketFactory(builder.build(), null, null,
+			SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER )
+	clientBuilder.setSSLSocketFactory(sslConnectionSocketFactory)
+	httpClient = clientBuilder.build()
+	return httpClient
+}
+	
+//* This script request an application process deployment
+ run(args)
+
+
+ def deploy ( String url, String user, String password, String application,
+			 String applicationProcess, String environment, String deployVersions, boolean disableSSLVerify, boolean verbose) {
+	
+	println "**  Deploying component versions: $deployVersions"
+	println "*** Starting deployment process '$applicationProcess' of application '$application' in environment '$environment'"
+
+	def rc = 0
+	def urlString= "$url/cli/applicationProcessRequest/request"
+	def httpClient = getHttpClient(disableSSLVerify)
+	def request = new HttpPut(urlString)
+	request.addHeader(new BasicHeader("Content-Type", "application/json"))
+	
+	
+	HttpClientContext clientContext = HttpClientContext.create();
+	
+	CredentialsProvider provider = new BasicCredentialsProvider();
+	UsernamePasswordCredentials credentials = new UsernamePasswordCredentials("$user", "$password");
+	provider.setCredentials(AuthScope.ANY, credentials);
+	clientContext.setCredentialsProvider(provider);
+	
+	def jsonString = """
+{
+  "application": "$application",
+  "applicationProcess": "$applicationProcess",
+  "environment": "$environment",
+  "versions": [
+"""
+	
+	def deployVersionsList = deployVersions.split("\\n")
+	deployVersionsList.each { deployVersion ->
+		def items = deployVersion.split(":")
+		def component = items[0]
+		def version = items[1]
+		jsonString += """
+    {
+      "version": "$version",
+      "component": "$component"
+    },
+"""
+	}
+	
+	jsonString += "    ]\n}"
+	
+	if ( verbose)
+		println "*** Request payload:\n$jsonString"
+	
+	def entity =  new StringEntity(new String (jsonString, "UTF-8"))
+	request.setEntity(entity)
+	def response = httpClient.execute(request,clientContext)
+	def statusCode = response.getStatusLine().getStatusCode()
+	println "*** Status Code: $statusCode"
+	def responseString = EntityUtils.toString(response.getEntity())
+	if ( statusCode != 200 ) {
+		rc = 1
+		println("*** Deployment failed:\n")
+		println "$responseString"
+	} else {
+		Thread.sleep( 1000 )
+		JsonSlurper slurper = new groovy.json.JsonSlurper()
+		def json = slurper.parseText("$responseString")
+		def requestId = json["requestId"]
+		def requestUrl = "$url/#applicationProcessRequest/$requestId" 
+		println "*** Follow Process Request: $requestUrl"
+		request = new HttpGet("$url/cli/applicationProcessRequest/$requestId")
+		response = httpClient.execute(request,clientContext)
+		statusCode = response.getStatusLine().getStatusCode()
+		responseString = EntityUtils.toString(response.getEntity())
+		json = slurper.parseText("$responseString")
+		while (statusCode == 200 
+			&& ( json["state"] == "EXECUTING" || json["state"] == "INITIALIZED" ) ) {
```

Comment:
I think, we can also have a **PENDING** status, when the deployment has not yet kicked off.

https://www.ibm.com/support/knowledgecenter/SS4GSP_7.1.1/com.ibm.udeploy.api.doc/topics/rest_cli_applicationprocessrequest_requeststatus_get.html